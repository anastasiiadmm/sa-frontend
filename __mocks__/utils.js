import * as reduxHooks from 'react-redux';

jest.mock('../src/redux/store');
jest.mock('react-redux');

export const mockedUseSelectors = jest.spyOn(reduxHooks, 'useSelector');
export const mockedDispatch = jest.spyOn(reduxHooks, 'useDispatch');
