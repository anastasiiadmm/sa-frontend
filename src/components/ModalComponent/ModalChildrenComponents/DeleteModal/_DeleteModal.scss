@use 'assets/scss/_themes.scss';
@use 'assets/scss/_mixins.scss';

.DeleteModal {
  &__delete-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }

  &__delete-modal-buttons {
    margin-top: 25px;
    display: flex;
    flex-direction: row;
    gap: 10px;
    width: 100%;
  }

  &__delete-profile-button {
    background: themes.$red-color !important;
    &:hover {
      background: rgba(193, 16, 48, 0.6) !important;
    }
  }

  &__delete-image {
    width: 48px;
    height: 48px;
  }

  @include mixins.bp(mobile) {
    &__delete-modal-buttons {
      flex-direction: column;
      > button {
        height: 44px;
        border: none!important;
        border-radius: 8px!important;
      }
    }
    &__delete-container {
      display: flex;
      justify-content: left;
      align-items: normal;

      > h4 {
        margin-top: 5px;
        margin-bottom: 5px;
      }
      > span {
        text-align: left!important;
      }
    }
  }
}

@include mixins.bp(mobile) {
  .delete-drawer-block {
    padding: 30px;
  }
}
