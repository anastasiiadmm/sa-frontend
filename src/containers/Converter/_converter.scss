@use "assets/scss/_themes.scss";
@use "assets/scss/_mixins.scss";

.Converter {
  display: flex;
  flex-direction: column;
  gap: 12px;
  padding: 9px;

  &__converter-block {
    display: flex;
    gap: 15px;
  }

  &__card-file-block {
    width: 100%;

    > div {
      padding: 12px !important;
    }
  }

  &__converter-card {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-evenly;
    gap: 16px;
  }

  &__convert-item {
    width: 230px;
  }

  &__title {
    margin: 0;
    text-transform: uppercase;
  }

  &__select-block {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    margin-bottom: 20px;
  }

  &__history-list {
    display: flex;
    flex-wrap: wrap;
    background: themes.$white-color;
    width: 100%;
    border-radius: themes.$border-radius;
  }

  &__heading {
    margin: 0 !important;
    flex-grow: 1;
  }

  &__subtitle {
    flex-grow: 1;
  }

  &__close-button {
    border: none;
    &:hover {
      border: none;
    }
    box-shadow: 0 0 0 rgba(0, 0, 0, 0.02);
  }

  &__link-img {
    width: 35px;
    margin-bottom: 10px;
  }

  &__files-link-block {
    display: flex;
    flex-direction: column;
  }

  &__select-field {
    > div {
      border: none !important;
      background-color: themes.$light-gray-color !important;
    }
  }

  .ant-typography h5 {
    margin-bottom: 0 !important;
  }

  .title-heading {
    margin-top: 12px;
    padding-left: 2px;
  }

  @include mixins.bp(desktop) {
    &__history-list {
      display: flex;
      flex-wrap: wrap;
      background: none;
      width: auto;
      border-radius: initial;
    }
  }
}

.button-width {
  width: auto !important;
}
