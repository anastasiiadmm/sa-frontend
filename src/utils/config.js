export const apiURL = '';

export const defaultError = { detail: 'На сервере что-то пошло не так' };
