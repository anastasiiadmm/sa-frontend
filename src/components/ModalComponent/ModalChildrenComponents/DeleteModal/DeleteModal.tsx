import { Button, Typography } from 'antd';
import bem from 'easy-bem';
import React from 'react';

import deleteIcon from 'assets/images/icons/newIcon/warning.svg';
import 'components/ModalComponent/ModalChildrenComponents/DeleteModal/_DeleteModal.scss';

interface Props {
  handleDeleteCancel?: () => void;
  deleteButtonHandler?: () => void;
  loading?: boolean | { delay?: number | undefined } | undefined;
  title?: string;
  fullName: string;
}

const { Title, Text } = Typography;

const DeleteModal: React.FC<Props> = ({
  handleDeleteCancel,
  deleteButtonHandler,
  loading,
  fullName,
  title,
}) => {
  const b = bem('DeleteModal');

  return (
    <div className={b('delete-container')}>
      <img className={b('delete-image')} src={deleteIcon} alt='deleteIcon' />
      <Title level={4}>{title}</Title>
      <Text
        style={{
          textAlign: 'center',
        }}
      >
        Вы уверены, что хотите удалить <br /> <b>{fullName}</b>
      </Text>
      <div className={b('delete-modal-buttons')}>
        <Button
          type='primary'
          style={{ width: '100%', borderRadius: 4 }}
          className={b('cancel-profile-button')}
          onClick={handleDeleteCancel}
        >
          Отменить
        </Button>
        <Button
          type='primary'
          loading={loading}
          htmlType='submit'
          style={{ width: '100%', borderRadius: 4 }}
          className={b('delete-profile-button')}
          onClick={deleteButtonHandler}
        >
          Удалить
        </Button>
      </div>
    </div>
  );
};

export default DeleteModal;
