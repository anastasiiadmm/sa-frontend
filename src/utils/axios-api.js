import axios from "axios";
import {apiURL} from "utils/config";

export const axiosApi = axios.create({
  baseURL: apiURL,
});