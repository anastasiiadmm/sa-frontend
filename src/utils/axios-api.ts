import axios from 'axios';

import { checkForTokens, logoutUser } from 'redux/auth/authSlice';
import { apiURL } from 'utils/config';
import { logoutLocalStorage } from 'utils/token';
import store from 'redux/store';

const axiosApi = axios.create({
  baseURL: apiURL,
});

axiosApi.interceptors.request.use(async (config) => {
  const key = store.getState()?.auth?.tokens?.access;

  if (key) {
    config.headers = {
      ...config.headers,
      Authorization: `Bearer ${key}`,
      'Content-Type': 'application/json',
    } as any;
  }
  return config;
});

axiosApi.interceptors.response.use(
  async (config) => {
    return config;
  },
  async (error) => {
    const originalRequest = error.config;

    const statusCode = error?.response?.status;
    const index = store;
    const { tokens } = index.getState().auth;
    if (
      tokens &&
      statusCode === 401 &&
      error.config &&
      !error.config._isReady &&
      error.response.data.messages
    ) {
      originalRequest._isReady = true;
      try {
        const resp = await axiosApi.post('/accounts/refresh/', { refresh: tokens.refresh });
        if (resp.status === 200) {
          const newTokens = resp.data;
          axiosApi.defaults.headers.Authorization = `Bearer ${newTokens.access}`;
          const usersLocal = {
            user: index.getState()?.auth?.user,
            token: {
              access: resp.data.access,
              refresh: tokens.refresh,
            },
          };
          index.dispatch(checkForTokens(usersLocal));
          const obj: any = usersLocal;
          delete obj.token.refresh;
          localStorage.setItem('users', JSON.stringify(obj));
          window.dispatchEvent(new Event('storage'));
          return axiosApi(originalRequest);
        }
      } catch (e) {
        logoutLocalStorage();
        window.location.reload();
      }

      store.dispatch(logoutUser());
    }

    return Promise.reject(error);
  },
);

export default axiosApi;
