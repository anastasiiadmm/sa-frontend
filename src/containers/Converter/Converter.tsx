import { CloseOutlined } from '@ant-design/icons';
import { Button, Card, message, Spin, Typography } from 'antd';
import bem from 'easy-bem';
import moment from 'moment';
import React, { useEffect, useRef, useState } from 'react';
import { Link } from 'react-router-dom';

import arrowDown from 'assets/images/icons/arrow-down.svg';
import converterFile from 'assets/images/icons/converter-file.svg';
import Errors from 'components/Errors/Errors';
import FormField from 'components/FormField/FormField';
import DeleteModal from 'components/ModalComponent/ModalChildrenComponents/DeleteModal/DeleteModal';
import ModalComponent from 'components/ModalComponent/ModalComponent';
import PaginationComponent from 'components/TableComponent/PaginationComponent/PaginationComponent';
import useWindowWidth from 'hooks/useWindowWidth';
import { IConverter } from 'interfaces/IConverter';
import {
  clearConvertFileSuccess,
  converterSelector,
  convertFile,
  deleteConverter,
  fetchConverterList,
} from 'redux/converter/converterSlice';
import { useAppDispatch, useAppSelector } from 'redux/hooks';
import { convertOptions } from 'utils/constants';
import {
  dateWithTimeFormat,
  dateWithTimeSecFormat,
  downloadConvertedFileHandler,
  getPageNumber,
  getPageNumberPrevious,
} from 'utils/helper';
import 'containers/Converter/_converter.scss';

const { Title, Text } = Typography;

const Converter = () => {
  const b = bem('Converter');
  const dispatch = useAppDispatch();
  const fileInputRef = useRef<HTMLInputElement | null>(null);
  const {
    converterList,
    converterListPagination,
    converterListLoading,
    converterListError,
    deleteConverterLoading,
    deleteConverterError,
    convertFileSuccess,
  } = useAppSelector(converterSelector);
  const [isModalOpen, setIsModalOpen] = useState(false);
  const windowWidth = useWindowWidth();
  const [filters, setFilters] = useState({
    page: converterListPagination?.next
      ? Number(getPageNumber(converterListPagination?.next))
      : Number(getPageNumberPrevious(converterListPagination?.previous)),
  });
  const [fileName, setFileName] = useState<{ id: number | null; name: string }>({
    id: null,
    name: '',
  });
  const [isLoading, setIsLoading] = useState<{ [key: string]: boolean }>({});
  const [file, setFile] = useState<File | null>(null);

  useEffect(() => {
    return () => {
      dispatch(clearConvertFileSuccess());
    };
  }, [dispatch, convertFileSuccess]);

  useEffect(() => {
    const data = {
      query: {
        page: filters?.page || 1,
      },
    };
    dispatch(fetchConverterList({ data }));
  }, [dispatch, filters]);

  const pagePrevHandler = () => {
    setFilters({ ...filters, page: filters.page - 1 });
  };

  const pageNextHandler = () => {
    setFilters({ ...filters, page: filters.page + 1 });
  };

  const handleButtonClick = () => {
    if (fileInputRef.current) {
      fileInputRef.current.click();
    }
  };

  const handleDownloadClick = (file: string) => {
    setIsLoading((prevIsLoading) => ({ ...prevIsLoading, [file]: true }));
    downloadConvertedFileHandler(file, () =>
      setIsLoading((prevIsLoading) => ({ ...prevIsLoading, [file]: false })),
    );
  };

  const showDeleteModal = (id: number, name: string) => {
    setIsModalOpen(true);
    setFileName({ id, name });
  };

  const handleDeleteOkCancel = () => {
    setIsModalOpen(!isModalOpen);
  };

  const onFileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const files = event.target.files;
    if (files) {
      setFile(files[0]);
    }
  };

  const clearFileHandle = () => {
    setFile(null);
    if (fileInputRef.current) {
      fileInputRef.current.value = '';
    }
  };

  const handleConvertHandler = async () => {
    try {
      const formData = new FormData();
      if (file) {
        formData.append('file', file);
      }
      await dispatch(convertFile(formData)).unwrap();
      await clearFileHandle();
      message.success('Файл успешно сконвертирован');
    } catch (e) {
      if (e?.detail?.non_field_errors) {
        const errorMessage = e?.detail?.non_field_errors[0];

        if (errorMessage === 'Archive file does not contain supported file format') {
          await message.error('Архивный файл не содержит поддерживаемого формата файла');
        }
      } else {
        await message.error('Произошла ошибка');
      }
    }
  };

  const deleteHandler = async () => {
    try {
      await dispatch(deleteConverter(fileName?.id)).unwrap();
      setFileName({ id: null, name: '' });
      setIsModalOpen(!isModalOpen);
    } catch (e) {
      await message.error(`${e?.detail}`);
    }
  };

  if (windowWidth >= 990 && (converterListError || deleteConverterError)) {
    return <Errors status={converterListError?.status} detail={converterListError?.detail} />;
  }

  return (
    <>
      <div className={b('')} data-testid='converter-test-id'>
        <Card>
          <div className={b('converter-block')}>
            <div>
              <img src={converterFile} alt={converterFile} />
            </div>
            <div className={b('convert-item')}>
              <Title className={b('title')} level={3}>
                Конвертер
              </Title>
              <p>Сконвертируйте ваши файлы в любой формат</p>
              <div className={b('select-block')}>
                из{' '}
                <FormField
                  dropdownStyle={{ display: 'none' }}
                  defaultValue={convertOptions[0]}
                  className={b('select-field')}
                  type='select'
                  customStyle='80px'
                  suffixIcon={<img src={arrowDown} alt='arrowDown' />}
                />{' '}
                в{' '}
                <FormField
                  dropdownStyle={{ display: 'none' }}
                  defaultValue={convertOptions[1]}
                  className={b('select-field')}
                  type='select'
                  customStyle='80px'
                  suffixIcon={<img src={arrowDown} alt='arrowDown' />}
                />
              </div>
              {file ? (
                <div className={b('file-info-block')}>
                  <Text strong>{file?.name}</Text>
                  <Button size='small' icon={<CloseOutlined />} onClick={clearFileHandle} />
                </div>
              ) : null}
              {file ? (
                <Button type='primary' className='button-style' onClick={handleConvertHandler}>
                  Конвертировать
                </Button>
              ) : (
                <Button className='button-style' onClick={handleButtonClick}>
                  Выбрать файл
                </Button>
              )}

              <input
                ref={fileInputRef}
                data-testid='image-input'
                id='image-input'
                type='file'
                onChange={onFileChange}
              />
            </div>
          </div>
        </Card>

        {windowWidth <= 990 ? (
          <Link to='/files'>
            <Card hoverable style={{ width: 170 }}>
              <div className={b('files-link-block')}>
                <img src={converterFile} alt='converterFile' className={b('link-img')} />
                <Text strong>Мои файлы</Text>
              </div>
            </Card>
          </Link>
        ) : (
          <>
            <Text className={b('title title-heading')}>История конвертаций</Text>
            {converterListLoading ? (
              <Spin />
            ) : (
              <div>
                <div className={b('history-list')}>
                  {converterList?.map((converter: IConverter) => {
                    const isLoadingState = isLoading[converter?.file];

                    return (
                      <Card className={b('card-file-block')} key={converter?.id}>
                        <div className={b('converter-card')}>
                          <div className={b('info-column')}>
                            <Title className={b('heading')} level={5}>
                              {converter?.task_UID && converter?.file
                                ? converter.task_UID +
                                  converter.file.substring(converter.file.lastIndexOf('.'))
                                : converter?.task_UID}
                            </Title>
                          </div>
                          <div className={b('info-column')}>
                            <div className={b('info-item')}>
                              <Text type='secondary' className={b('subtitle')}>
                                {moment(converter?.created_at, dateWithTimeSecFormat).format(
                                  dateWithTimeFormat,
                                )}
                              </Text>
                            </div>
                            <div className={b('info-item')}>
                              <Button
                                className='button-style button-width'
                                loading={isLoadingState}
                                onClick={() => handleDownloadClick(converter?.file)}
                              >
                                Скачать
                              </Button>
                              <Button
                                className='button-style button-width'
                                danger
                                onClick={() => showDeleteModal(converter?.id, converter?.task_UID)}
                              >
                                Удалить
                              </Button>
                            </div>
                          </div>
                        </div>
                      </Card>
                    );
                  })}
                </div>
                <PaginationComponent
                  params={
                    converterListLoading
                      ? { previous: null, next: null, count: 0 }
                      : converterListPagination
                  }
                  pagePrevHandler={pagePrevHandler}
                  pageNextHandler={pageNextHandler}
                />
              </div>
            )}
          </>
        )}
      </div>

      <ModalComponent
        dividerShow={false}
        open={isModalOpen}
        handleOk={handleDeleteOkCancel}
        handleCancel={handleDeleteOkCancel}
      >
        <DeleteModal
          title='Удалить?'
          loading={deleteConverterLoading}
          fullName={fileName?.name}
          handleDeleteCancel={handleDeleteOkCancel}
          deleteButtonHandler={deleteHandler}
        />
      </ModalComponent>
    </>
  );
};

export default Converter;
