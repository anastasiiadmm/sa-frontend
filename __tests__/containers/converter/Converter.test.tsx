import { BrowserRouter } from "react-router-dom";
import { cleanup, fireEvent, render, screen, waitFor } from "@testing-library/react";
import "../../../__mocks__/matchMedia.mock";
import "@testing-library/jest-dom";
import { mockedDispatch, mockedUseSelectors } from "../../../__mocks__/utils";
import Converter from "../../../src/containers/Converter/Converter";
import DeleteModal from "../../../src/components/ModalComponent/ModalChildrenComponents/DeleteModal/DeleteModal";

afterEach(cleanup);

const data = [
  {
    id: 91,
    created_at: '21/08/2023 16:11:30+0600',
    task_UID: 'taskUID_1692612690.2575638',
    file: '/media/files/converter/taskUID_1692612690.2575638.txt'
  },
  {
    id: 92,
    created_at: '26/08/2023 19:11:30+0600',
    task_UID: 'taskUID_1692612690.2575635',
    file: '/media/files/converter/taskUID_1692612690.2575635.txt'
  }
];

describe("<Converter />", () => {
  let handleDeleteCancel: jest.Mock;
  let deleteUserHandler: jest.Mock;

  beforeEach(() => {
    handleDeleteCancel = jest.fn();
    deleteUserHandler = jest.fn();
  });

  test("Converter list component should be in the document", async () => {
    mockedUseSelectors.mockReturnValue(data);
    const dispatch = jest.fn();
    mockedDispatch.mockReturnValue(dispatch);

    render(
      <BrowserRouter>
        <Converter />
      </BrowserRouter>
    );

    const converterListComponent = screen.getByTestId("converter-test-id");

    await waitFor(() => {
      expect(converterListComponent).toBeInTheDocument();
    });
  });

  test("render component toMatchSnapshot()", async () => {
    mockedUseSelectors.mockReturnValue(data);
    const dispatch = jest.fn();
    mockedDispatch.mockReturnValue(dispatch);

    const utils = render(
      <BrowserRouter>
        <Converter />
      </BrowserRouter>
    );

    expect(utils).toMatchSnapshot();
  });

  test("Converted file should set to modal form success", async () => {
    const { getByText } = render(
      <DeleteModal
        fullName='21/08/2023 16:11:30+0600'
        handleDeleteCancel={handleDeleteCancel}
        deleteButtonHandler={deleteUserHandler}
        loading={false}
      />
    );

    fireEvent.click(getByText('Удалить'));
    await waitFor(() => expect(deleteUserHandler).toHaveBeenCalledTimes(1));
  });
});
