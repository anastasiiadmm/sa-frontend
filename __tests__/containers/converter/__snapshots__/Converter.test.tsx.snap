// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Converter /> render component toMatchSnapshot() 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Converter"
        data-testid="converter-test-id"
      >
        <div
          class="ant-card ant-card-bordered css-dev-only-do-not-override-15rg2km"
        >
          <div
            class="ant-card-body"
          >
            <div
              class="Converter__converter-block"
            >
              <div>
                <img
                  alt="files"
                  src="[object Object]"
                />
              </div>
              <div
                class="Converter__convert-item"
              >
                <h3
                  class="ant-typography Converter__title css-dev-only-do-not-override-15rg2km"
                >
                  Конвертер
                </h3>
                <p>
                  Сконвертируйте ваши файлы в любой формат
                </p>
                <div
                  class="Converter__select-block"
                >
                  из
                   
                  <div
                    class="ant-select Converter__select-field css-dev-only-do-not-override-15rg2km ant-select-single ant-select-show-arrow"
                    style="width: 80px;"
                  >
                    <div
                      class="ant-select-selector"
                    >
                      <span
                        class="ant-select-selection-search"
                      >
                        <input
                          aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                          aria-autocomplete="list"
                          aria-controls="rc_select_TEST_OR_SSR_list"
                          aria-expanded="false"
                          aria-haspopup="listbox"
                          aria-owns="rc_select_TEST_OR_SSR_list"
                          autocomplete="off"
                          class="ant-select-selection-search-input"
                          id="rc_select_TEST_OR_SSR"
                          readonly=""
                          role="combobox"
                          style="opacity: 0;"
                          type="search"
                          unselectable="on"
                          value=""
                        />
                      </span>
                      <span
                        class="ant-select-selection-item"
                        title=".zip"
                      >
                        .zip
                      </span>
                    </div>
                    <span
                      aria-hidden="true"
                      class="ant-select-arrow"
                      style="user-select: none;"
                      unselectable="on"
                    >
                      <img
                        alt="arrowDown"
                        src="[object Object]"
                      />
                    </span>
                  </div>
                   
                  в
                   
                  <div
                    class="ant-select Converter__select-field css-dev-only-do-not-override-15rg2km ant-select-single ant-select-show-arrow"
                    style="width: 80px;"
                  >
                    <div
                      class="ant-select-selector"
                    >
                      <span
                        class="ant-select-selection-search"
                      >
                        <input
                          aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                          aria-autocomplete="list"
                          aria-controls="rc_select_TEST_OR_SSR_list"
                          aria-expanded="false"
                          aria-haspopup="listbox"
                          aria-owns="rc_select_TEST_OR_SSR_list"
                          autocomplete="off"
                          class="ant-select-selection-search-input"
                          id="rc_select_TEST_OR_SSR"
                          readonly=""
                          role="combobox"
                          style="opacity: 0;"
                          type="search"
                          unselectable="on"
                          value=""
                        />
                      </span>
                      <span
                        class="ant-select-selection-item"
                        title=".txt"
                      >
                        .txt
                      </span>
                    </div>
                    <span
                      aria-hidden="true"
                      class="ant-select-arrow"
                      style="user-select: none;"
                      unselectable="on"
                    >
                      <img
                        alt="arrowDown"
                        src="[object Object]"
                      />
                    </span>
                  </div>
                </div>
                <button
                  class="ant-btn css-dev-only-do-not-override-15rg2km ant-btn-default button-style"
                  type="button"
                >
                  <span>
                    Выбрать файл
                  </span>
                </button>
                <input
                  data-testid="image-input"
                  id="image-input"
                  type="file"
                />
              </div>
            </div>
          </div>
        </div>
        <span
          class="ant-typography Converter__title title-heading css-dev-only-do-not-override-15rg2km"
        >
          История конвертаций
        </span>
        <div>
          <div
            class="Converter__history-list"
          />
          <div
            class="pagination undefined"
          >
            <div
              class="pagination__pagination_show"
            >
              <div
                class="glav_pagination"
              >
                <div
                  class="next_table"
                >
                  <button
                    class="ant-btn css-dev-only-do-not-override-15rg2km ant-btn-default ant-btn-icon-only pagination_buttons pagination_buttons-prev"
                    disabled=""
                    type="button"
                  >
                    <span
                      class="ant-btn-icon"
                    >
                      <img
                        alt="backIcons"
                        src="[object Object]"
                      />
                    </span>
                  </button>
                  <button
                    class="ant-btn css-dev-only-do-not-override-15rg2km ant-btn-default ant-btn-icon-only pagination_buttons pagination_buttons-next"
                    disabled=""
                    role="button"
                    type="button"
                  >
                    <span
                      class="ant-btn-icon"
                    >
                      <img
                        alt="nextIcons"
                        src="[object Object]"
                      />
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Converter"
      data-testid="converter-test-id"
    >
      <div
        class="ant-card ant-card-bordered css-dev-only-do-not-override-15rg2km"
      >
        <div
          class="ant-card-body"
        >
          <div
            class="Converter__converter-block"
          >
            <div>
              <img
                alt="files"
                src="[object Object]"
              />
            </div>
            <div
              class="Converter__convert-item"
            >
              <h3
                class="ant-typography Converter__title css-dev-only-do-not-override-15rg2km"
              >
                Конвертер
              </h3>
              <p>
                Сконвертируйте ваши файлы в любой формат
              </p>
              <div
                class="Converter__select-block"
              >
                из
                 
                <div
                  class="ant-select Converter__select-field css-dev-only-do-not-override-15rg2km ant-select-single ant-select-show-arrow"
                  style="width: 80px;"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                        aria-autocomplete="list"
                        aria-controls="rc_select_TEST_OR_SSR_list"
                        aria-expanded="false"
                        aria-haspopup="listbox"
                        aria-owns="rc_select_TEST_OR_SSR_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="rc_select_TEST_OR_SSR"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-item"
                      title=".zip"
                    >
                      .zip
                    </span>
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <img
                      alt="arrowDown"
                      src="[object Object]"
                    />
                  </span>
                </div>
                 
                в
                 
                <div
                  class="ant-select Converter__select-field css-dev-only-do-not-override-15rg2km ant-select-single ant-select-show-arrow"
                  style="width: 80px;"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                        aria-autocomplete="list"
                        aria-controls="rc_select_TEST_OR_SSR_list"
                        aria-expanded="false"
                        aria-haspopup="listbox"
                        aria-owns="rc_select_TEST_OR_SSR_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="rc_select_TEST_OR_SSR"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-item"
                      title=".txt"
                    >
                      .txt
                    </span>
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <img
                      alt="arrowDown"
                      src="[object Object]"
                    />
                  </span>
                </div>
              </div>
              <button
                class="ant-btn css-dev-only-do-not-override-15rg2km ant-btn-default button-style"
                type="button"
              >
                <span>
                  Выбрать файл
                </span>
              </button>
              <input
                data-testid="image-input"
                id="image-input"
                type="file"
              />
            </div>
          </div>
        </div>
      </div>
      <span
        class="ant-typography Converter__title title-heading css-dev-only-do-not-override-15rg2km"
      >
        История конвертаций
      </span>
      <div>
        <div
          class="Converter__history-list"
        />
        <div
          class="pagination undefined"
        >
          <div
            class="pagination__pagination_show"
          >
            <div
              class="glav_pagination"
            >
              <div
                class="next_table"
              >
                <button
                  class="ant-btn css-dev-only-do-not-override-15rg2km ant-btn-default ant-btn-icon-only pagination_buttons pagination_buttons-prev"
                  disabled=""
                  type="button"
                >
                  <span
                    class="ant-btn-icon"
                  >
                    <img
                      alt="backIcons"
                      src="[object Object]"
                    />
                  </span>
                </button>
                <button
                  class="ant-btn css-dev-only-do-not-override-15rg2km ant-btn-default ant-btn-icon-only pagination_buttons pagination_buttons-next"
                  disabled=""
                  role="button"
                  type="button"
                >
                  <span
                    class="ant-btn-icon"
                  >
                    <img
                      alt="nextIcons"
                      src="[object Object]"
                    />
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
